src/router.js

```diff
@@ -452,13 +452,21 @@ function validateRouteConfig(config: RouteConfig, routes: Array<Object>): void {
 }
 
 function evaluateNavigationStrategy(instruction: NavigationInstruction, evaluator: Function, context: any): Promise<NavigationInstruction> {
-  return Promise.resolve(evaluator.call(context, instruction)).then(() => {
+  return Promise.resolve(evaluator.call(context, instruction)).then((modules?: string | {[viewportname: string]: moduleId}) => {
     if (!('viewPorts' in instruction.config)) {
-      instruction.config.viewPorts = {
-        'default': {
-          moduleId: instruction.config.moduleId
-        }
-      };
+      instruction.config.viewPorts = {};
+    }
+
+    if (typeof modules === 'string') {
+      modules = {'default': modules};
+    } else if (modules === undefined) {
```

Comment:
For example, if someone has logic in their nav strat that falls back to a null value, they might expect it to run with the default config, but instead, modules will remain null and this will throw an unhandled error.

I'd recommend updating this to check for a falsey value.